src/transformers/pipelines.py

```diff
@@ -2058,6 +2057,267 @@ def span_to_answer(self, text: str, start: int, end: int) -> Dict[str, Union[str
+        )
+
+        self.aggregate = bool(getattr(self.model.config, "aggregation_labels")) and bool(
+            getattr(self.model.config, "num_aggregation_labels")
+        )
+
+    def batch_inference(self, **inputs):
+        return self.model(**inputs)
+
+    def sequential_inference(self, **inputs):
+        """
+        Inference used for models that need to process sequences in a sequential fashion, like the SQA models which
+        handle conversational query related to a table.
+        """
+        all_logits = []
+        all_aggregations = []
+        prev_answers = None
+        batch_size = inputs["input_ids"].shape[0]
+
+        input_ids = inputs["input_ids"].to(self.device)
+        attention_mask = inputs["attention_mask"].to(self.device)
+        token_type_ids = inputs["token_type_ids"].to(self.device)
+        token_type_ids_example = None
+
+        for index in range(batch_size):
+            # If sequences have already been processed, the token type IDs will be created according to the previous
+            # answer.
+            if prev_answers is not None:
+                coords_to_answer = prev_answers[index]
+                prev_label_ids_example = token_type_ids_example[:, 3]  # shape (seq_len,)
+                model_label_ids = np.zeros_like(prev_label_ids_example.cpu().numpy())  # shape (seq_len,)
+
+                token_type_ids_example = token_type_ids[index]  # shape (seq_len, 7)
+                for i in range(model_label_ids.shape[0]):
+                    segment_id = token_type_ids_example[:, 0].tolist()[i]
+                    col_id = token_type_ids_example[:, 1].tolist()[i] - 1
+                    row_id = token_type_ids_example[:, 2].tolist()[i] - 1
+
+                    if row_id >= 0 and col_id >= 0 and segment_id == 1:
+                        model_label_ids[i] = int(coords_to_answer[(col_id, row_id)])
+
+                token_type_ids_example[:, 3] = torch.from_numpy(model_label_ids).type(torch.long).to(self.device)
+
+            input_ids_example = input_ids[index]
+            attention_mask_example = attention_mask[index]  # shape (seq_len,)
+            token_type_ids_example = token_type_ids[index]  # shape (seq_len, 7)
+            outputs = self.model(
+                input_ids=input_ids_example.unsqueeze(0),
+                attention_mask=attention_mask_example.unsqueeze(0),
+                token_type_ids=token_type_ids_example.unsqueeze(0),
+            )
+            logits = outputs.logits
+
+            if self.aggregate:
+                all_aggregations.append(outputs.logits_aggregation)
+
+            all_logits.append(logits)
+
+            dist_per_token = torch.distributions.Bernoulli(logits=logits)
+            probabilities = dist_per_token.probs * attention_mask_example.type(torch.float32).to(
+                dist_per_token.probs.device
+            )
+
+            coords_to_probs = collections.defaultdict(list)
+            prev_answers = {}
```

Comment:
is this correct here? Init empty `prev_answers` dict only to fill it with a single value after? I think there is a problem with the `prev_answers` dict here - If I read the code correctly I don't think `prev_answers` has to be a dict since it's overwritten at each iteration of the loop. Can it maybe just be a variable? Or move this statement somewhere else?